src/app/shared/search/search-result.component.ts

```diff
@@ -89,12 +91,44 @@ export class SearchResultComponent implements OnChanges {
       this.resultView.columns = this.resultView.columns.sort((a: SearchResultViewColumn, b: SearchResultViewColumn) => {
         return a.order - b.order;
       });
+
+      this.hydrateResultView();
     }
     if (changes['page']) {
       this.selected.page = (changes['page']).currentValue;
     }
   }
 
+  /**
+   * Hydrates result view with case field definitions.
+   */
+  // A longer term resolution is to move this piece of logic to the backend
+  hydrateResultView(): void {
+    this.resultView.results.forEach(result => {
+      const caseFields = [];
+
+      Object.keys(result.case_fields).forEach(fieldId => {
+
+        const field = result.case_fields[fieldId];
+        caseFields.push({
+          id: fieldId,
+          label: null,
+          field_type: {},
+          value: field,
+          display_context: null,
+        });
+      });
+
+      result.hydrated_case_fields = caseFields;
```

Comment:
can we find a better naming than hydrated_case_fields? it's difficult to understand without looking at the code what's the purpose of this array of CaseFields

Reply:
what's the reason for not having here: 
```result.hydrated_case_fields = result.case_fields```
I'm sure there is a reason but can't get it :)

Reply:
`hydrated_case_fields` is an array, while `result.case_fields` is an object used as a map. 
Because `ccdLabelSubstitutor` is expecting an array of fields, not a map, we're converting it to an array.

Reply:
and `result.case_fields` only contains the raw values, not CaseField instances like `ccdLabelSubstitutor` is expecting

Reply:
I see. What about a method 
getCaseFieldsArray(): CaseField[] on SearchResultViewItem itself then?